casanovo/casanovo.py

```diff
@@ -489,6 +507,58 @@ def _get_model_weights(cache_dir: Path) -> str:
             )
 
 
+def _setup_output(
+    output_dir: str | None,
+    output_root: str | None,
+    overwrite: bool,
+    verbosity: str,
+) -> Tuple[Path, str]:
+    """
+    Set up the output directory, output file root name, and logging.
+
+    Parameters:
+    -----------
+    output_dir : str | None
+        The path to the output directory. If `None`, the output directory will
+        be resolved to the current working directory.
+    output_root : str | None
+        The base name for the output files. If `None` the output root name will
+        be resolved to casanovo_<current date and time>
+    overwrite: bool
+        Whether to overwrite log file if it already exists in the output
+        directory.
+    verbosity : str
+        The verbosity level for logging.
+
+    Returns:
+    --------
+    Tuple[Path, str]
+        A tuple containing the resolved output directory and root name for
+        output files.
+    """
+    if output_root is None:
+        output_root = (
+            f"casanovo_{datetime.datetime.now().strftime('%Y%m%d%H%M%S')}"
+        )
+
+    if output_dir is None:
+        output_path = Path.cwd()
+    else:
+        output_path = Path(output_dir).expanduser().resolve()
+        if not output_path.is_dir():
+            output_path.mkdir(parents=True)
+            warnings.warn(
```

Comment:
**todo:** This should be `logging.warn(...)` instead.

Conceptually you can consider `warnings.warn` messages for _developers_ and `logging.xxx` messages for users.